catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/ModalRegistry.view.js

```diff
@@ -131,6 +125,34 @@ function (ich,Marionette,Backbone,ConfigurationEdit,wreqr,_,$,Utils,Service,moda
                     });
             }
         },
+        urlChanged: function(evt) {
+            var url = evt.target.value;
+            var regExUrl;
+            if (url) {
+                regExUrl = this.extractHostnameFromUrl(url);
+            }
+            var hostname;
+            if (regExUrl) {
+                hostname = regExUrl;
+                if (hostname) {
+                    this.checkName(hostname);
+                } else {
+                    this.checkName(url);
+                }
+            }
+            var $registryName = this.$(".registryName");
+            var $group = this.$el.find('.registryName>.control-group');
+            $group.empty();
+            $registryName.val(hostname);
+        },
+        extractHostnameFromUrl: function(url){
+            var myRe = new RegExp("^(?:([^:/?#.]+):)?(?://)?(([^:/?#]*)(?::(\\d*))?)((/(?:[^?#](?![^?#/]*\\.[^?#/.]+(?:[\\?#]|$)))*/?)?([^?#/]*))?(?:\\?([^#]*))?(?:#(.*))?");
```

Comment:
I don't know how exactly, but I feel like there might be a way to get the hostname without having to resort to regex parsing.